flow/e2e/s3/cdc_s3_test.go

```diff
@@ -80,3 +80,52 @@ func (s PeerFlowE2ETestSuiteS3) Test_Complete_Simple_Flow_S3() {
 	env.Cancel(s.t.Context())
 	e2e.RequireEnvCanceled(s.t, env)
 }
+
+func (s PeerFlowE2ETestSuiteS3) Test_OriginMetadata() {
+	tc := e2e.NewTemporalClient(s.t)
+
+	srcTableName := s.attachSchemaSuffix("origin_metadata")
+	dstTableName := "peerdb_test_s3.origin_metadata"
+	flowJobName := s.attachSuffix("origin_metadata")
+
+	_, err := s.conn.Conn().Exec(s.t.Context(), fmt.Sprintf(`
+		CREATE TABLE %s (
+			id SERIAL PRIMARY KEY,
+			val TEXT NOT NULL
+		);
+	`, srcTableName))
+	require.NoError(s.t, err)
+	connectionGen := e2e.FlowConnectionGenerationConfig{
+		FlowJobName:      flowJobName,
+		TableNameMapping: map[string]string{srcTableName: dstTableName},
+		Destination:      s.Peer().Name,
+	}
+
+	flowConnConfig := connectionGen.GenerateFlowConnectionConfigs(s)
+	flowConnConfig.Env = map[string]string{"PEERDB_ORIGIN_METADATA_AS_DESTINATION_COLUMN": "true"}
+
+	env := e2e.ExecutePeerflow(s.t.Context(), tc, peerflow.CDCFlowWorkflow, flowConnConfig, nil)
+	e2e.SetupCDCFlowStatusQuery(s.t, env, flowConnConfig)
+	// insert 20 rows
+	for i := range 20 {
+		testKey := fmt.Sprintf("test_key_%d", i)
+		testValue := fmt.Sprintf("test_value_%d", i)
+		_, err = s.conn.Conn().Exec(s.t.Context(), fmt.Sprintf(`
+			INSERT INTO %s (key, val) VALUES ($1, $2)
+		`, srcTableName), testKey, testValue)
+		e2e.EnvNoError(s.t, env, err)
+	}
+	e2e.EnvNoError(s.t, env, err)
+
+	e2e.EnvWaitFor(s.t, env, 2*time.Minute, "waiting for blobs", func() bool {
+		ctx, cancel := context.WithTimeout(s.t.Context(), 25*time.Second)
+		defer cancel()
+		files, err := s.s3Helper.ListAllFiles(ctx, flowJobName)
+		e2e.EnvNoError(s.t, env, err)
+		s.t.Logf("Files in Test_Complete_Simple_Flow_S3 %s: %d", flowJobName, len(files))
+		return len(files) == 4
+	})
```

Comment:
this tests that we write something to s3, but ideally test pulls avro from s3 & checks for columns

maybe kafka test would be better place, could also test scripting then